src/assets/src/components/common.tsx

```diff
@@ -274,38 +271,45 @@ interface SingleInputFormShowRemainingProps extends StatelessSingleInputFormProp
 }
 
 export const SingleInputFormShowRemaining: React.FC<SingleInputFormShowRemainingProps> = (props) => {
-    const inputRef = createRef<HTMLTextAreaElement>();
-    useEffect(() => {
-        if (!props.autofocus) return;
-        inputRef.current!.focus();
-    }, []);
-    const submit = (e: React.FormEvent<HTMLFormElement>) => {
-        e.preventDefault();
-        props.onSubmit(props.value);
-        props.onChangeValue("");
+    const [remaining, setRemaining] = useState(props.maxLength - props.value.length as number)
+    const [isInvalid, setIsInvalid] = useState(false as boolean)
+    const buttonClass = `btn btn-${props.buttonType} remaining-controls` as string
```

Comment:
No need to type-assert `as number`/`as boolean`/`as string` here as they can be inferred from the context.

Reply:
Sure, removed.